### PR TITLE
PARQUET-197 : fix parquet-cascading not writing parquet metadata file

### PR DESCRIPTION
**Pull request by tsdeng**
Repro: run a scalding job that writes parquet files to a folder. no _metadata and _common_metadata file is created
Impact: potential performance problem if parquet metadata is read from client side, which is the case for sparkSQL
casue: the metatdata writing logic is in the mapreduce API but not the mapred API of parquet.

### CODE REVIEW

File: parquet-hadoop/src/main/java/parquet/hadoop/mapred/DeprecatedParquetOutputCommitter.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import java.io.IOException;
+
+/**
+ *
+ * Adapter for supporting ParquetOutputCommitter in mapred API
+ *
+ * @author Tianshuo Deng
+ */
+public class DeprecatedParquetOutputCommitter extends FileOutputCommitter {
```
Comment (julienledem) [inline parquet-hadoop/src/main/java/parquet/hadoop/mapred/DeprecatedParquetOutputCommitter.java]: I'd prefer we say "MapredParquetOutputCommitter"
using "Deprecated" to refer to the deprecated but still heavily in use mapred api is confusing to users. "mapred" is better qualifier.

**Comment by julienledem** [comment]
Just one comment otherwise this looks good to me (assuming the build passes).
Please check why the build fails

**Comment by tsdeng** [comment]
Thanks @julienledem , I just updated the PR with adding a new method setAsOutputFormat to set both the output format class and output committer. Use this method then we don't forget to set the ParquetOutputCommiter when using DeprecatedParquetOutputFormat. Could you take a look again with the new changes?

**Comment by julienledem** [comment]
LGTM.
+1

**Comment by isnotinvain** [comment]
+1, thanks!

**Comment by isnotinvain** [comment]
testing here: https://github.com/isnotinvain/incubator-parquet-mr/pull/3

**Comment by isnotinvain** [comment]
Tested failed:
https://s3.amazonaws.com/archive.travis-ci.org/jobs/53133792/log.txt

**Pull request closed without merging**